File: Java/sdk/src/main/java/vdo/ai/sample/java/sdk/application/MyApplication.java
package vdo.ai.sample.java.sdk.application;

import android.app.Application;
import android.content.Context;

import androidx.annotation.Nullable;

import vdo.ai.android.core.VdoAppOpenAd;
import vdo.ai.android.core.listeners.VdoAppOpenListener;
import vdo.ai.android.core.models.VdoAdError;
import vdo.ai.android.sdk.VdoAIAppOpen;
import vdo.ai.sample.java.sdk.BuildConfig;


public class MyApplication extends Application {

    public VdoAIAppOpen appOpenAd;
    public static String IRON_SOURCE_APP_KEY ="85460dcd"; // sample app key

    @Override
    public void onCreate() {
        super.onCreate();


        appOpenAd = new VdoAIAppOpen.VdoAIAppOpenBuilder()
                .withContext(this)
                .setEnvironment(BuildConfig.BUILD_TYPE)
                .setTagName("in-app-sample-ap-Z1")
                .setApplovinAdUnitId("83c31eac2dd321a2")
                .setIronSourceParams(IRON_SOURCE_APP_KEY, "DefaultOfferWall")
                .setListener(new VdoAppOpenListener(){

                    @Override
                    public void onAdFailedToShowFullScreenContent(@Nullable VdoAdError vdoAdError) {

                    }

                    @Override
                    public void onAdFailedToLoad(@Nullable VdoAdError vdoAdError) {

                    }

                    @Override
                    public void onAdLoaded() {

                    }

                    @Override
                    public void onAdDismissedFullScreenContent() {

                    }

                    @Override
                    public void onAdShowedFullScreenContent() {

                    }
                }).build();
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        // if using MultiDexApplication not required Z1MediaManager.initMultiDex function
    }
}

File: Kotlin/example-sdk/src/main/java/vdo/ai/android/example/sdk/application/MyApplication.kt
package vdo.ai.android.example.sdk.application

import android.app.Application
import android.content.Context
import android.util.Log
import vdo.ai.android.core.listeners.VdoAppOpenListener
import vdo.ai.android.core.models.VdoAdError
import vdo.ai.android.example.sdk.BuildConfig
import vdo.ai.android.example.sdk.ironSourceOfferApiKey
import vdo.ai.android.sdk.VdoAIAppOpen

/**
 *  created by Ashish Saini at 105h Oct 2023
 *
 **/
class MyApplication : Application() {

    var appOpenAd:VdoAIAppOpen?=null

    override fun onCreate() {
        super.onCreate()

        appOpenAd = VdoAIAppOpen.VdoAIAppOpenBuilder()
            .withContext(this@MyApplication)
            .setEnvironment(BuildConfig.BUILD_TYPE)
            .setTagName("in-app-sample-ap-Z1")
            .setApplovinAdUnitId("83c31eac2dd321a2")
            .setIronSourceParams(ironSourceOfferApiKey, "DefaultOfferWall")
            .setListener(object : VdoAppOpenListener {
                override fun onAdLoaded() {
                }

                override fun onAdFailedToLoad(adError: VdoAdError?) {
                }

                override fun onAdDismissedFullScreenContent() {
                }

                override fun onAdShowedFullScreenContent() {
                }

                override fun onAdFailedToShowFullScreenContent(adError: VdoAdError?) {
                }
            }).build()

    }

    override fun attachBaseContext(context: Context) {
        super.attachBaseContext(context)
//        Z1MediaManager.initMultiDex(this)
    }
}

